tests/Moq.Analyzers.Test/CallbackSignatureShouldMatchMockedMethodAnalyzerTests.cs

```diff
@@ -0,0 +1,126 @@
+
+        // Invalid patterns that SHOULD trigger the analyzer
+        IEnumerable<object[]> invalidPatterns = new object[][]
+        {
+            // Basic callback with wrong parameter type
+            ["""new Mock<IFoo>().Setup(x => x.DoWork("test")).Callback(({|Moq1100:int wrongParam|}) => { });"""],
+
+            // Ref parameter mismatch (missing ref)
+            ["""new Mock<IFoo>().Setup(m => m.DoRef(ref It.Ref<string>.IsAny)).Callback(({|Moq1100:string data|}) => { });"""],
+
+            // Out parameter mismatch (missing out)
+            ["""new Mock<IFoo>().Setup(m => m.DoOut(out It.Ref<int>.IsAny)).Callback(({|Moq1100:int result|}) => { });"""],
+        }.WithNamespaces().WithMoqReferenceAssemblyGroups();
+
+        return validPatterns.Concat(invalidPatterns);
+    }
+
+    [Theory]
+    [MemberData(nameof(CallbackValidationData))]
+    public async Task ShouldValidateCallbackPatterns(string referenceAssemblyGroup, string @namespace, string testCode)
+    {
+        static string Template(string ns, string code) =>
+            $$"""
+            {{ns}}
+
+            public interface IFoo
+            {
+                int DoWork(string input);
+                bool ProcessMultiple(int id, string name, DateTime timestamp);
+                void ProcessData(ref string data);
+                bool TryProcess(out int result);
+                void ProcessMixed(int id, ref string data, out bool success);
+                void ProcessReadOnly(in DateTime timestamp);
+                int DoRef(ref string data);
+                bool DoOut(out int result);
+                string DoIn(in DateTime timestamp);
+                T ProcessGeneric<T>(T input);
+            }
+
+            public class TestClass
+            {
+                public void TestMethod()
+                {
+                    {{code}}
+                }
+            }
+            """;
+
+        string source = Template(@namespace, testCode);
+        await AnalyzerVerifier.VerifyAnalyzerAsync(source, referenceAssemblyGroup);
+    }
+
+    /// <summary>
+    /// Test to document the current limitation with generic callback validation.
+    /// This test documents that .Callback&lt;T&gt;() with wrong type parameters is NOT currently validated.
+    /// This could be enhanced in a future version.
+    /// </summary>
+    /// <returns>A task representing the asynchronous unit test.</returns>
+    [Fact]
+    public async Task GenericCallbackValidation_CurrentLimitation_IsDocumented()
+    {
+        const string source = """
+            using Moq;
+
+            public interface IFoo
+            {
+                int DoWork(string input);
+            }
+
+            public class TestClass
+            {
+                public void TestGenericCallback()
+                {
+                    var mock = new Mock<IFoo>();
+                    // Note: This currently does NOT trigger a diagnostic, which could be enhanced in the future
```

Comment:
@coderabbitai Make note of this gap. Reference the Moq documentation and create a new issue to document what the pattern is, why it's a failure case that requires an analyzer, and reference documentation. Use INVEST